scRNA-seq-advanced/04-differential_expression.Rmd

```diff
@@ -12,156 +12,255 @@ output:
-To help us out later on when we are performing our analysis and making plots, let's incorporate the information in this sample metadata table into the `colData` of the integrated dataset.
+To help us out later on when we are performing our analysis and making plots, let's incorporate the information in this sample metadata table into the `colData` of the integrated SCE object.
+This will allow us to match each of the samples in the SCE object with the diagnosis group.
 
-```{r}
-# add the sample metadata to the coldata 
+```{r modify coldata}
+# add the sample metadata to the colData from the integrated SCE object
 coldata_df <- colData(integrated_sce) %>%
+  # convert from DataFrame to data.frame
   as.data.frame() %>%
   # merge with sample metadata 
   dplyr::left_join(sample_metadata, by = c("sample" = "library_id")) %>%
-    # simplify subdiagnosis for easy plotting and DE later
-  dplyr::mutate(cell_id = paste(sample, barcode, sep = "-"),
+  # create a new columns
+  dplyr::mutate(cell_id = paste(sample, barcode, sep = "-"), # combination of barcode and sample
+                # simplify subdiagnosis for easy plotting and DE later
                 diagnosis_group = forcats::fct_recode(
                   subdiagnosis,
                   "ARMS" = "Alveolar rhabdomyosarcoma",
                   "ERMS" = "Embryonal rhabdomyosarcoma"
                 ))
 
-# add back to sce 
+# add modified data.frame back to SCE as DataFrame
 colData(integrated_sce) <- DataFrame(coldata_df, row.names = coldata_df$cell_id)
+
+# take a look at the new modified colData
+head(colData(integrated_sce))
 ```
 
-Now we can use that sample metadata when making our plots. 
-Lets start by looking at our different subdiagnoses.
+Now when we look at the `colData` of the SCE object we should see new columns, including the `diagnosis_group` column which indicates if each cell comes from an ERMS or ARMS sample.
+We can now use that column to label any UMAP plots (or other plot types) that we make.
+In the below chunk we will color our cells by subdiagnosis.
 
-```{r}
+```{r diagnosis group UMAP}
+# UMAP of all samples, separating by diagnosis group
 scater::plotReducedDim(integrated_sce,
                        dimred = "fastmnn_UMAP",
-                       colour_by = "sample",
+                       colour_by = "diagnosis_group",
                        point_size= 0.5,
-                       point_alpha = 0.2,
-                       other_fields = "diagnosis_group") +
-  facet_wrap(~diagnosis_group)
+                       point_alpha = 0.2) 
 ```
 
-Interestingly, it looks like samples from the ARMS and ERMS subdiagnoses tend to integrate with each other rather than integrate all together. 
+Interestingly, it looks like samples from the ARMS and ERMS subdiagnoses tend to integrate with or samples of the same subdiagnosis rather than integrate all together. 
```

Comment:
extra "or"? 

I might phrase this more as "subdiagnoses tend to cluster together"